geth/peers/peerpool_test.go

```diff
@@ -166,13 +166,13 @@ func (s *PeerPoolSimulationSuite) TestSingleTopicDiscoveryWithFailover() {
 
 	s.Equal(signal.EventDiscoveryStarted, s.getPoolEvent(poolEvents))
 	s.Require().NotNil(s.peers[1].DiscV5)
-	register = NewRegister(topic)
```

Comment:
i didnt get it :) you disabled topic advertisement for 2nd peer. this is exactly second part of this test

Reply:
@dshulyak Yeah, avoiding re-creating this register again ([already created here](https://github.com/status-im/status-go/pull/984/files#diff-bba072cf08ae14911b1c4657643a88deR139)), it reduces a lot the error rates. But I'm still trying understand whats going on.

BTW, any suggestion will be welcome :)

Reply:
I'm still not sure, but the fact is you get much less fails with that line commented. 
I'll spend some time this morning trying to debug and figure out the why of this behaviour

Reply:
i tried locally and it works even with cpu=1, let's merge it then i guess

Reply:
oh, i think it was just a coincidence that it passed locally. i still can make it fail with -cpu=1

Reply:
yeah, it doesn't avoid it from failing, it just reduces how often it fails

Reply:
indeed it even worked on my other PR which was constantly failing with this error

Reply:
maybe we can merge this small change while we look for a better solution, broken builds are a PITA

Reply:
yes 100%